### ISSUE TITLE
storage: `TestPebbleEquivalence` failure with missing provisional value

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#86088]
Seen in https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/6085907:

```
=== RUN   TestPebbleEquivalence/engine/random=9
check file = /artifacts/tmp/_tmp/3cb480549dd56d78f96a16434f955825/TestPebbleEquivalence781029557/output.meta
output file = /artifacts/tmp/_tmp/3cb480549dd56d78f96a16434f955825/TestPebbleEquivalence_engine_random=93802187071/output.meta
store path = /artifacts/tmp/_tmp/3cb480549dd56d78f96a16434f955825/TestPebbleEquivalence_engine_random=91609265826/store
    generator.go:416: mismatching output at line 17440, operation index 17339: expected truncated range to delete = "cjgkvgibuyyki" - "xpfycslgsrjwn", deleted keys = "embhqahjh", "flsfvwghkgejel", "gktmrfwbwct", "gqxwpvnsbguuex", "gwskfeqqy", "hffnogymjssf", "hmbfclosekyugyw", "iycswlqadu", "jfnvtlfhfx", "jtaqkcmrrs", "keipluyanygoy", "kghguysqflwm", "kiefrxpoptesm", "kpyxwlvcg", "kxtbfrlyaakvgl", "mvchkvqs", "ncqrrbxgmgrom", "ntsqcufujgy", "ollexsutk", "ormiquujbi", "ovenbvunigkhbu", "ovslrpxv", "pcgmrgjmxt", "pgbykqbeanuyolf", "phlpccau", "pinpjablyqjvxl", "prkqfdyhyphnn", "pschisvcyfok", "pswptibhbhxiul", "tbefuffod", "umlcrsurkwf", "unhghawobtyrh", "vdydlwsuaqknc", "vfxsfogo", "vhgxyjwjrv", "vhysqeoimpxuu", "viocigwkt", "vkkknwxwtrm", "vlolluedgydvf", "vmjftftoj", "vmjkugbry", "vovxikvrf", "vqxnjihpobcwp", "vrxtpinvtmln", "vufbytnkimovx", "vunjtxmkegk", "vxgdslqqk", "wajixogdibkhf", "waqtsqcmy", "wbkhwvuwsjlpd", "wcbliomlpcoviwg", "wcxbsebclm", "wdnhvslp", "wdsdqokejbmrmfn", "whlrlptyyd", "whtyjpjx", "wjrbdkgqivyv", "wktfvxrwkqrtvt", "wpnivvfwkwn", "wpyeneni", "wqsgqnkskbcgky", "wuspfmmiitm", "wykvglkydffm", "xaoeywuwlhv", "xdloqwej", "xekdyssmerlmjo", "xhahaxgbvd", "xjsemortvf", "xkcrugbtnx", "xlkdchxwwwhdvo", "xlkutumqlukslwx", "xniunwovahu", "xnoonduanlyg"
        , got error: iter not on provisional value for intent "pswptibhbhxiul"
    --- FAIL: TestPebbleEquivalence/engine/random=9 (13.46s)
```

Note that this is the old `DeleteRange` operation which drops point tombstones, not the new one using MVCC range tombstones.

Will pull artifacts once build completes.

Jira issue: CRDB-18565

Epic CRDB-2624

**Comment by erikgrinaker** [comment]
[TestPebbleEquivalence.tar.gz](https://github.com/cockroachdb/cockroach/files/9332028/TestPebbleEquivalence.tar.gz)

**Comment by erikgrinaker** [comment]
Seeing more failures like these over on this:

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/6086832?showRootCauses=true&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true

**Comment by erikgrinaker** [comment]
Bisected this down to #86064.

**Comment by erikgrinaker** [comment]
Fixed by #86234.
